lib/interceptor/dns.js

```diff
@@ -336,12 +336,15 @@ module.exports = interceptorOpts => {
 
   const instance = new DNSInstance(opts)
 
-  return dispatch => {
+  return (dispatch, opts) => {
     return function dnsInterceptor (origDispatchOpts, handler) {
-      const origin =
-        origDispatchOpts.origin.constructor === URL
-          ? origDispatchOpts.origin
-          : new URL(origDispatchOpts.origin)
+      if (!origDispatchOpts.origin) {
```

Comment:
Shall we remove it, now that dispatchers can be assumed they'll submit the origin?

Reply:
Yes, I'll do a new commit in a few hours to remove unnecessary stuff.